ibis/expr/rules.py

```diff
@@ -294,12 +295,41 @@ def _promote_integral_binop(exprs, op):
     return dt.highest_precedence(dtypes)
 
 
+def _infer_decimal_dtype(args, op):
+    # TODO: Add support for the maximum precision and maximum scale
+    # TODO: Currently assumes that there are only two args. Can there be more?
+    if len(args) != 2:
+        return highest_precedence_dtype(args)
+    lhs_prec = args[0].type().precision
+    lhs_scale = args[0].type().scale
+    rhs_prec = args[1].type().precision
+    rhs_scale = args[1].type().scale
+    if op is operator.mul:
+        return dt.Decimal(
+            lhs_prec + rhs_prec,
+            lhs_scale + rhs_scale,
+        )
+    if op is operator.add or op is operator.sub:
```

Comment:
Should I make a follow up issue to address division? It looks like that'll require max precision/scale as well.

Reply:
It seems that the output type for division is computed differently within ibis. For `/` it's *always* float 64, for `//` it's always int64. I was not sure why this works this way, so I left division out for the moment.

Reply:
I think we could implement the thing with max_prec and max_scale within this PR as well, if you prefer. I am just not certain where to put it within ibis.

Reply:
Ok, that makes sense. I'll make a follow-up issue. Longer term we should probably split out the arithmetic operations into a few subclasses based type category to simplify the type derivation logic. We could then address revision.

Reply:
#4334 is the follow up issue.